### PR TITLE
Cut scikit-image dependency

### PR DESCRIPTION
**Pull request by scottstanie**
The `resize` function is a thin wrapper around `ndimage.zoom`. There were only a few names changed.

The add an anti-aliasing filter, which might make more of a difference in other cases, but the SET is so smooth that it doesn't matter for us.  The difference is approximately 0:

image

### CODE REVIEW

**Comment by scottstanie** [comment]
I was also thinking of removing `matplotlib-base` as a dependency, since it's really just used to inspect the outputs. What do you think of that @yunjunz ?

File: src/pysolid/grid.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 import os
 
 import numpy as np
-from skimage.transform import resize
+import scipy.ndimage as ndi
```
Comment (yunjunz) [inline src/pysolid/grid.py:17]:
```suggestion
from scipy import ndimage
```

File: src/pysolid/grid.py [hunk 2]
```diff
@@ -80,12 +80,11 @@ def calc_solid_earth_tides_grid(dt_obj, atr, step_size=1e3, display=False, verbo
...
     if num_step > 1:
+        in_shape = tide_e.shape
         out_shape = (int(atr['LENGTH']), int(atr['WIDTH']))
         vprint('PYSOLID: resize data to the shape of {} using order-1 spline interpolation'.format(out_shape))
-        kwargs = dict(order=1, mode='edge', anti_aliasing=True, preserve_range=True)
-        tide_e = resize(tide_e, out_shape, **kwargs)
-        tide_n = resize(tide_n, out_shape, **kwargs)
-        tide_u = resize(tide_u, out_shape, **kwargs)
+        kwargs = dict(order=1, mode="nearest", grid_mode=True)
+        ndi.zoom(np.stack([tide_e, tide_n, tide_u]), [1, *np.divide(out_shape, in_shape)], **kwargs)
```
Comment (yunjunz) [inline src/pysolid/grid.py:87]: The resized `tide_e/n/u` are not returned. Could you fix this?

Comment (yunjunz) [reply src/pysolid/grid.py:87]: I am curious about the syntax of `[1, *np.divide(out_shape, in_shape)]`, it works nicely. Is there a document on it?

**Review by yunjunz — CHANGES_REQUESTED** [review]

**Comment by yunjunz** [comment]
> I was also thinking of removing `matplotlib-base` as a dependency, since it's really just used to inspect the outputs. What do you think of that @yunjunz ?

`matplotlib-base` is definitely optional and could have been removed from the conda-forge recipe at least. I am hesitant to remove it from the `requirements.txt` because people may find it confusing as they thought they have install the code but could not run the notebook or example code on the readme. `matplotlib` is so widely used, is there a specific reason we want to not installing it?

File: src/pysolid/grid.py (hunk 2 above)
Comment (scottstanie) [reply src/pysolid/grid.py:87]: this is a form of unpacking arguments: https://docs.python.org/3/tutorial/controlflow.html#unpacking-argument-lists https://treyhunner.com/2018/10/asterisks-in-python-what-they-are-and-how-to-use-them/
like how you can call a function `f(a, b, c)` with 
```python
later_args = (2, 3)
f('blah', *later_args)
```
you can make a list with `['blah', *later_args]`.

If it seems "too clever", i can break it out to two lines

Comment (scottstanie) [reply src/pysolid/grid.py]: and thanks for catching that!

**Comment by scottstanie** [comment]
> matplotlib is so widely used, is there a specific reason we want to not installing it?

The specific reason was to try and make downstream packages that use PySolid (like `COMPASS`) smaller.

If I take the `environment.yml` file in this PR and make a [lockfile with all the dependencies using this script](https://github.com/opera-adt/dolphin/blob/423da2dfd90db8ef3f299be1226b6708203ae4c6/docker/create-lockfile.sh), the total number of conda packages goes from 67 to 173
```
$ ../dolphin/docker/create-lockfile.sh environment.yml > pysolid_with_matplotlib.txt
# Then remove matplotlib-base in `environment-no-mpl.yml`
$  ../dolphin/docker/create-lockfile.sh environment-no-mpl.yml > pysolid_no_matplotlib.txt

$ wc -l pysolid_*_matplotlib.txt
   67 pysolid_no_matplotlib.txt
  173 pysolid_with_matplotlib.txt
```

And building when docker image with those two environments fies, the one with matplotlib is about double in size the one without:

```
REPOSITORY          TAG               IMAGE ID       CREATED          SIZE
insarlab/pysolid    with-matplotlib   d90399eaefdb   25 seconds ago   1.85GB
insarlab/pysolid    no-matplotlib     a8ae2dc6fbd3   57 seconds ago   934MB
```

So in practice you're right that for a lot of people who are going to install pysolid into some big all-encompasing environment, it makes no difference. but if (and when) isce3 splits off from `NISAR` and cuts down it's dependencies, it'll be nice for environemnts to solve faster and docker images to be smaller.
We can push off that change to another PR that also would update documentation if you'd like to keep this one focused

**Comment by hfattahi** [comment]
> > matplotlib is so widely used, is there a specific reason we want to not installing it?
> 
> The specific reason was to try and make downstream packages that use PySolid (like `COMPASS`) smaller.
> 
> If I take the `environment.yml` file in this PR and make a [lockfile with all the dependencies using this script](https://github.com/opera-adt/dolphin/blob/423da2dfd90db8ef3f299be1226b6708203ae4c6/docker/create-lockfile.sh), the total number of conda packages goes from 67 to 173
> 
> ```
> $ ../dolphin/docker/create-lockfile.sh environment.yml > pysolid_with_matplotlib.txt
> # Then remove matplotlib-base in `environment-no-mpl.yml`
> $  ../dolphin/docker/create-lockfile.sh environment-no-mpl.yml > pysolid_no_matplotlib.txt
> 
> $ wc -l pysolid_*_matplotlib.txt
>    67 pysolid_no_matplotlib.txt
>   173 pysolid_with_matplotlib.txt
> ```
> 
> And building when docker image with those two environments fies, the one with matplotlib is about double in size the one without:
> 
> ```
> REPOSITORY          TAG               IMAGE ID       CREATED          SIZE
> insarlab/pysolid    with-matplotlib   d90399eaefdb   25 seconds ago   1.85GB
> insarlab/pysolid    no-matplotlib     a8ae2dc6fbd3   57 seconds ago   934MB
> ```
> 
> So in practice you're right that for a lot of people who are going to install pysolid into some big all-encompasing environment, it makes no difference. but if (and when) isce3 splits off from `NISAR` and cuts down it's dependencies, it'll be nice for environemnts to solve faster and docker images to be smaller. We can push off that change to another PR that also would update documentation if you'd like to keep this one focused

We definitely need to remove matplotlib for obvious reasons that @scottstanie mentioned above. It actually came up already on data system side. Up to you guys if you want to do that in this PR or a separate PR.

File: src/pysolid/grid.py (hunk 2 above)
Comment (yunjunz) [reply src/pysolid/grid.py]: Thanks for the explanation, I love it!

**Review by yunjunz — APPROVED** [review]
Looks all good to me. Thank you @scottstanie!

**Comment by yunjunz** [comment]
Removing `matplotlib` in another PR sounds good to me. I like your `tests/requirements.txt` file in `dolphin`, so we could add `matplotlib` there. 

Regarding the dependency files, I believe `environment.yml` file is redundant, we could remove it as well.

**Comment by yunjunz** [comment]
The CI looked fine to me. It usually takes ~1 hour to finish all the testing on various Linux versions.

**Comment by scottstanie** [comment]
Oh whoops! Sorry about cancelling then

**Pull request merged**